digdag-tests/src/test/java/acceptance/RequireIT.java

```diff
@@ -185,6 +203,159 @@ public void testIgnoreProjectIdParam()
+        {
+            String sessionTime = "2020-05-28 12:34:21";
+            RestSessionAttemptCollection childAttempts = testRerunOnParam(sessionTime, "failed", true);
+            assertThat("Number of child's attempt must be two. (== require> kick child)", childAttempts.getAttempts().size(), is(2));
+        }
+
+        // test for rerun_on: failed. previous child attempt succeeded
+        {
+            String sessionTime = "2020-05-28 12:34:31";
+            RestSessionAttemptCollection childAttempts = testRerunOnParam(sessionTime, "failed", false);
+            assertThat("Number of child's attempt must be one. (== require> kick)", childAttempts.getAttempts().size(), is(1));
+        }
+    }
+
+    private RestSessionAttemptCollection testRerunOnParam(String sessionTime, String rerunOn, boolean previousChildRunFail) throws InterruptedException
+    {
+        //Start a child
+        String childFailParam = previousChildRunFail ? "yes" : "no";
+        CommandStatus childStatus = startAndWait(true,"require", "child", "--session", sessionTime,
+                "-p", "child_fail=" + childFailParam);
+        assertThat(isAttemptSuccess(childStatus), not(previousChildRunFail));
+
+        // Start a parent with same session time.
+        CommandStatus rerunOnNoneStatus = startAndWait(false,"require", "parent", "--session", sessionTime,
+                "-p", "param_rerun_on=" + rerunOn,
+                "-p", "child_fail=no");
+
+        assertThat(isAttemptSuccess(rerunOnNoneStatus), is(true));
+        RestSessionAttemptCollection attempts = client.getSessionAttemptRetries(getAttemptId(childStatus));
+        return attempts;
+    }
+
+    private static boolean isAttemptSuccess(CommandStatus status) { return status.outUtf8().contains("status: success"); }
+
+    private CommandStatus startAndWait(boolean ignoreAttemptFailure, String... args) throws InterruptedException
+    {
+        List<String> newArgs = new ArrayList<>(Arrays.asList("start",
+                "-c", config.toString(),
+                "-e", server.endpoint()
+                ));
+        newArgs.addAll(Arrays.asList(args));
+        CommandStatus startStatus = main(newArgs);
+        Id startAttemptId = getAttemptId(startStatus);
+        CommandStatus attemptsStatus = null;
+        // Wait for the attempt to complete
+        boolean success = false;
+        for (int i = 0; i < 30; i++) {
+            attemptsStatus = main("attempt",
+                    "-c", config.toString(),
+                    "-e", server.endpoint(),
+                    String.valueOf(startAttemptId));
+            if (attemptsStatus.outUtf8().contains("status: success")) {
+                success = true;
+                break;
+            }
+            else if (attemptsStatus.outUtf8().contains("status: error")) {
+                break;
+            }
+            Thread.sleep(1000);
+        }
+        if (!ignoreAttemptFailure) {
+            assertThat(success, is(true));
```

Comment:
Oh, the same check is done above like `assertThat(isAttemptSuccess(childStatus), not(previousChildRunFail))`?
So we can remove this to make things simpler?

Reply:
Thanks for your pointing out. I will remove them and argument ignoreAttemptFailure.

Reply:
Fixed it and refactoring. Thanks for your pointing out.